en/identity-server/next/docs/references/extend/user-mgt/write-a-custom-event-handler.md

```diff
@@ -0,0 +1,136 @@
+# Write a Custom Event Handler
+
+The WSO2 Identity Server eventing framework can be used to trigger events such as user operation events like `PRE_SET_USER_CLAIMS`,`POST_ADD_USER`. A full list of the sample events can be found below. The eventing framework also supports handlers which can be used to do operations upon a triggered event. For instance, an event handler can be used to validate the changed user password against previously used entries when a `PRE_UPDATE_CREDENTIAL` event is triggered. 
+
+---
+
+## What is an event handler?
+
+An event handler is used to perform an operation based on the published events. 
+
+For instance, an event handler can be used to send an email after a user addition. The following sequence of operations are executed while adding a user.
+
+1. Publish the `PRE_ADD_USER` event — The subscribed handlers will be executed for the pre-add user event.
+2. Execute the `AddUser` operation — The user will be persisted to the user store (LDAP or JBDC user store).
+3. Publish the `POST_ADD_USER` event — The subscribed handlers will be executed for the post-add user event.
+
+Therefore, the email can be sent through an event handler that is subscribed to the `POST_ADD_USER` event.
+
+The following list is a list of sample events. 
+
+- `AUTHENTICATION_STEP_SUCCESS`
+- `AUTHENTICATION_STEP_FAILURE`
+- `AUTHENTICATION_SUCCESS`
+- `AUTHENTICATION_FAILURE`
+- `PRE_AUTHENTICATION`
+- `POST_AUTHENTICATION`
+- `PRE_SET_USER_CLAIMS`
+- `POST_SET_USER_CLAIMS`
+- `PRE_ADD_USER`
+- `POST_ADD_USER`
+
+!!! info
+    The other events available with WSO2 Identity Server can be found from the `Event` class [here](https://github.com/wso2/carbon-identity-framework/blob/master/components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/IdentityEventConstants.java).
+
+
+---
+
+## Write an event handler
+
+To write a new event handler, you must extend the `org.wso2.carbon.identity.event.handler.AbstractEventHandler`. 
+
+1. Override the `getName()` method to set the name for the event handler and the `getPriority()` method can be used to set the priory of the event handler. The handlers will be executed based on the priority.
+
+    ```java
+    public String getName() {
+    return "customEventHandler";
```

Comment:
```suggestion
         return "customEventHandler";
```
fix indentation for better view

Reply:
Fixed with 117f6a81c92f846c04e4193c37344c0df46f5e55